librz/core/cmd/cmd_print.c

```diff
@@ -6599,3 +6382,219 @@ RZ_IPI RzCmdStatus rz_print_byte_bitstream_handler(RzCore *core, int argc, const
+		case 2:
+			pj_i(state->d.pj, value);
+			break;
+		case 4:
+		case 8:
+			pj_n(state->d.pj, value);
+			break;
+		default:
+			rz_warn_if_reached();
+			break;
+		}
+		pj_ks(state->d.pj, "string", str);
+		pj_end(state->d.pj);
+		break;
+	}
+	default:
+		rz_warn_if_reached();
+		break;
+	}
+}
+
+static bool print_value(RzCore *core, PrintValueOptions *opts, RzCmdStateOutput *state) {
+	ut64 old_at = core->offset;
+	ut8 *block = core->block;
+	int blocksize = core->blocksize;
+	ut8 *block_end = core->block + blocksize;
+
+	if (block + 8 >= block_end) {
+		RZ_LOG_ERROR("core: block is truncated.\n");
+		return false;
+	}
+	ut64 repeat = opts->repeat;
+	if (opts->useBytes && opts->size > 0 && repeat > 0) {
+		repeat /= opts->size;
+	}
+	ut64 at = old_at;
+	rz_cmd_state_output_array_start(state);
+	do {
+		rz_core_seek(core, at, false);
+
+		ut64 v = 0;
+		switch (opts->size) {
+		case 1:
+			v = rz_read_ble8(block);
+			block += opts->size;
+			break;
+		case 2:
+			v = rz_read_ble16(block, core->print->big_endian);
+			block += opts->size;
+			break;
+		case 4:
+			v = rz_read_ble32(block, core->print->big_endian);
+			block += opts->size;
+			break;
+		case 8:
+			v = rz_read_ble64(block, core->print->big_endian);
+			block += opts->size;
+			break;
+		case 0:
+			v = rz_read_ble64(block, core->print->big_endian);
+			opts->size = core->rasm->bits / 8;
+			switch (core->rasm->bits / 8) {
+			case 1: v &= UT8_MAX; break;
+			case 2: v &= UT16_MAX; break;
+			case 4: v &= UT32_MAX; break;
+			case 8: v &= UT64_MAX; break;
+			default: break;
+			}
+			block += core->rasm->bits / 8;
+			break;
+		}
+		print_value_single(core, opts, at, v, state);
+		repeat--;
+		at += opts->size;
+	} while (repeat > 0);
+	rz_cmd_state_output_array_end(state);
+	rz_core_seek(core, old_at, false);
+	return true;
+}
+
+RZ_IPI RzCmdStatus rz_print_value_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	int repeat = argc > 1 ? rz_num_math(core->num, argv[1]) : 1;
+	if (repeat < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PrintValueOptions opts = {
+		.size = 0,
+		.repeat = repeat,
+		.useBytes = false
+	};
+	print_value(core, &opts, state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_value1_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	int repeat = argc > 1 ? rz_num_math(core->num, argv[1]) : 1;
+	if (repeat < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PrintValueOptions opts = {
+		.size = 1,
+		.repeat = repeat,
+		.useBytes = false
+	};
+	print_value(core, &opts, state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_value2_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	int repeat = argc > 1 ? rz_num_math(core->num, argv[1]) : 1;
+	if (repeat < 0) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PrintValueOptions opts = {
+		.size = 2,
+		.repeat = repeat,
+		.useBytes = false
+	};
+	print_value(core, &opts, state);
+	return RZ_CMD_STATUS_OK;
```

Comment:
this can be extracted in a function that just accept `size` as argument

Reply:
Done